File: Practice/ABC/Bcontest050_b.cpp
#include <bits/stdc++.h>
using namespace std;

#define rep(i, n) for (int i = 0; i < (int)(n); i++)

int main() {
    int N, M, P, X;
    long long ans=0, gap;

    cin >> N;
    vector<long long> T(N), ANSER(N);
    rep(i, N){
        cin >> T.at(i);
        ans += T[i];
    }

    cin >> M;
    rep(i, M) {
        cin >> P >> X;
        gap = T[P-1] - X;
        ANSER.at(i) = ans - gap;
    }

    rep(i, M) {
        cout << ANSER[i] << endl;
    }


    return 0;
}

File: Practice/ABC/Bcontest050_b.py
N = int(input())
T = list(map(int, input().split()))
sum_ = sum(T)
M = int(input())
ans = list()
for i in range(M) :
    P, X = map(int,input().split())
    gap = T[P-1] - X
    ans.append(sum_ - gap)

for i in range(len(ans)):
    print(ans[i])
